src/crud/document.py

```
@@ -194,27 +241,70 @@ async def create_documents(
+                )
+
+            # Retry vector upsert with exponential backoff
+            try:
+                async for attempt in AsyncRetrying(
+                    stop=stop_after_attempt(3),
+                    wait=wait_exponential(multiplier=0.5, min=0.5, max=2.0),
+                    reraise=True,
+                ):
+                    with attempt:
+                        await vector_store.upsert_many(namespace, vector_records)
+            except Exception as e:
```

Ruff (0.14.7)

Do not catch blind exception: `Exception`

(BLE001)